examples/systemcalls/systemcalls.c

```diff
@@ -80,20 +106,67 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
 
+    /* Confirm that path provided is absolute */
+    if (command[0][0] != '/')
+    {
+        return false;
+    }
 
-/*
- * TODO
- *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
- *   redirect standard out to a file specified by outputfile.
- *   The rest of the behaviour is same as do_exec()
- *
-*/
+    /* Open file to redirect standard out */
+    /* Content inspired by https://stackoverflow.com/questions/13784269/redirection-inside-call-to-execvp-not-working/13784315#13784315 */
+    int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+    if (fd < 0)
+    {
+        return false;
+    }
 
-    va_end(args);
+    /* Call fork, returning false if there is an error in invocation */
+    pid_t pid = fork();    
```

Comment:
@kbiggs how could you reuse logic and avoid duplication with the function above?